### PR TITLE
read_parquet Does Not Throw Error for Missing Column

### PR DESCRIPTION
**Pull request by malachi-constant**
### Relates
- #1315 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: awswrangler/s3/_read_parquet.py [hunk 1]
```diff
@@ -481,6 +482,15 @@ def _read_parquet_file(
             read_dictionary=categories,
             coerce_int96_timestamp_unit=pyarrow_args["coerce_int96_timestamp_unit"],
         )
+        if validate_schema and pq_file:
+            pq_file_table: pyarrow.lib.Table = pq_file.read()
```
Comment (kukushking) [inline awswrangler/s3/_read_parquet.py:486]: Looks like you can directly do `pq_file.schema.column()` w/o having to `.read()` twice. Or you can read the file like on 496, perform validation after and then return

Comment (malachi-constant) [reply awswrangler/s3/_read_parquet.py:486]: @kukushking Looks like `.column()` only accepts an `int` not sure how to use that in this case.

```
>>> parquet_file.schema.column()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: column() takes exactly one argument (0 given)
>>> parquet_file.schema.column(0)
<ParquetColumnSchema>
  name: a
  path: a
  max_definition_level: 1
  max_repetition_level: 0
  physical_type: INT64
  logical_type: None
  converted_type (legacy): NONE
>>> parquet_file.schema.column('a')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "pyarrow/_parquet.pyx", line 706, in pyarrow._parquet.ParquetSchema.column
TypeErr
```

Comment (kukushking) [reply awswrangler/s3/_read_parquet.py:486]: There's `pq_file.schema.names` also, and pq_file.schema_arrow.filed(f)` - have a look at the [ParquetSchema](https://github.com/apache/arrow/blob/master/python/pyarrow/_parquet.pyx#L829) and arrow [Schema](https://arrow.apache.org/docs/python/generated/pyarrow.Schema.html#pyarrow.Schema) classes

Comment (malachi-constant) [reply awswrangler/s3/_read_parquet.py:486]: Ahh, I think `.schema.names` will be perfect, thanks @kukushking !

**Review by kukushking — APPROVED** [review]

**Pull request merged**